modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/README.md

```diff
@@ -0,0 +1,119 @@
+
+With search pipelines, the operator can combine multiple [search processors](#search-processors) to create a transform which acts on the search request and/or search response.
+
+The primary benefit of search pipelines is that they live entirely inside OpenSearch and therefore require less development time. Further, search pipelines support standard APIs for accessing metrics and disaster recovery.
+
+## Search Processors
+
+You can create many search pipelines by combining search processors in various orders. There are two types of search processors:
+
+1. search request processors which transform a request _before_ it is executed;
+2. search response processors which transform the output of a request _after_ it is executed.
+
+Currently, there are two search request and two search response processors.
+
+### Creating a search processor
+
+New search processors can be created by following these steps:
+
+1. Create a new class in `org.opensearch.search.pipeline.common`, this class will hold your new processor and should include whether it is a request or response processor. For example, a response processor which deleted a target field could be called `DeleteFieldResponseProcessor`.
+
+2. Make the class extend the generic `AbstractProcessor` class as well as implement either the `SearchRequestProcessor` or `SearchResponseProcessor` class depending on what type of processor it is. In the `DeleteFieldResponseProcessor` example, this would look like:
+
+
+  public class DeleteFieldResponseProcessor extends AbstractProcessor implements SearchResponseProcessor
+
+3. Create the main functionality of your processor and implement the methods required by the implemented interface. This will be `SearchRequest processRequest(SearchRequest request) throws Exception;` for a search request processor or `SearchResponse processResponse(SearchRequest request, SearchResponse response) throws Exception;` for a search response processor.
+
+For the example field `DeleteFieldResponseProcessor`, this will look like:
+
+```
+@Override
+public SearchResponse processResponse(SearchRequest request, SearchResponse response) throws Exception {
+boolean foundField = false;
+
+    SearchHit[] hits = response.getHits().getHits();
+    for (SearchHit hit : hits) {
+       
+      // Process each hit as desired 
+      }
+
+      if (hit.hasSource()) {
+        
+                // Change hit source if needed 
+        );
+
+        Map<String, Object> sourceAsMap = typeAndSourceMap.v2();
+        if (sourceAsMap.containsKey(field)) {
+          
+                // Handle source as map 
+        }
+      }
+
+      if (!foundField && !ignoreMissing) {
+        
+            // Handle error scenarios 
+      }
+    }
+
+    return response;
+}
+```
+
+4. Create a factory to encapsulate the behavior of the class so that processors can be handled uniformly.
```

Comment:
The factory is used to parse processor-specific JSON config parameters and use those to construct an instance of the processor.

In your DeleteFieldResponseProcessor example, you would probably want to take the name of the field to delete as a config parameter, I think.

Reply:
I see, that is my misunderstanding on what it was intended to do. I will correct that.